plugin-test/src/test/groovy/org/gradle/github/dependency/extractor/BaseExtractorTest.groovy

```diff
@@ -167,6 +150,51 @@ abstract class BaseExtractorTest extends Specification {
         return manifest
     }
 
+    protected List<String> getManifestNames() {
+        return jsonManifests().keySet() as List
+    }
+
+    protected GitHubManifest gitHubManifest(String manifestName) {
+        def jsonManifest = jsonManifest(manifestName)
+        return new GitHubManifest(jsonManifest)
+    }
+
+    protected static class GitHubManifest {
+        Map manifestData
+
+        GitHubManifest(Map manifestData) {
+            this.manifestData = manifestData
+        }
+
+        def getName() {
+            return manifestData.name
+        }
+
+        def getSourceFile() {
+            return (manifestData.file as Map).source_location
+        }
+
+        def getResolved() {
+            def resolved = manifestData.resolved as Map
+            return resolved.keySet() as List
+        }
+
+        def resolved(String key) {
+            def resolved = manifestData.resolved as Map<String, Map>
+            return resolved[key]
+        }
+
+        def checkResolved(String key, String packageUrl, Map checks = [:]) {
+            def resolved = manifestData.resolved as Map<String, Map>
+            def dep = resolved[key]
+            assert dep != null
+            assert dep.package_url == packageUrl
+            assert dep.relationship == (checks.relationship ?: "direct")
+            assert dep.dependencies == (checks.dependencies ?: [])
```

Comment:
As someone using these tests when I was writing them, I did find the `verifyAll` which would return all failures to be useful and decreased iteration time when there were multiple issues.  Especially because, in my experience, recompiling the `plugin` project, on my local machine, takes ~20 seconds. And the full test run takes ~1 min. Surfacing as many test failures as possible all at once reduces the number of test -> fix cycles you have to go through

Reply:
I haven't found this to be a problem in day-to-day development, and readability of tests is paramount for me.

In this particular case, I couldn't find a way to use `verifyAll`, which only seems to function within a Spock test method.